engine/particle/src/particle.cpp

```diff
@@ -2079,19 +2133,57 @@ namespace dmParticle
         stats->m_Time = i->m_PlayTime;
     }
 
-    uint32_t GetVertexBufferSize(uint32_t particle_count, ParticleVertexFormat vertex_format)
+    uint32_t GetVertexBufferSize(uint32_t particle_count, uint32_t vertex_size)
+    {
+        return particle_count * 6 * vertex_size;
+    }
+
+    uint32_t GetMaxVertexBufferSize(HParticleContext context, uint32_t vertex_size)
     {
-        uint32_t vertex_size = sizeof(Vertex);
+        return GetVertexBufferSize(context->m_MaxParticleCount, vertex_size);
+    }
 
-        if (vertex_format == PARTICLE_GUI)
-            vertex_size = sizeof(ParticleGuiVertex);
+    #define ATTRIBUTE_WRAPPER_SIZE (sizeof(float) * 16)
 
-        return particle_count * 6 * vertex_size;
+    // EDITOR ONLY
+    static void* AcquireAttributeDataPtr(HParticleContext context)
+    {
+        uint32_t ix = context->m_AttributeDataPtrIndex;
+        context->m_AttributeDataPtrIndex++;
+
+        if (ix >= context->m_AttributeDataPtrs.Capacity())
+        {
+            void* new_ptr = malloc(ATTRIBUTE_WRAPPER_SIZE);
+            context->m_AttributeDataPtrs.OffsetCapacity(1);
+            context->m_AttributeDataPtrs.Push(new_ptr);
+            return new_ptr;
+        }
```

Comment:
We need to retain the same pointers between every simulation step, which other expanding array solutions will not guarantee, and I don't think we want to preallocate a large array for this since the scratch buffer approach is ONLY for the editor.